tachyon/base/containers/container_util.h

```diff
@@ -47,6 +48,20 @@ std::vector<ReturnType> CreateVector(size_t size, Generator&& generator) {
   return ret;
 }
 
+template <typename Generator,
+          typename FunctorTraits = internal::MakeFunctorTraits<Generator>,
+          typename ReturnType = typename FunctorTraits::ReturnType,
+          typename RunType = typename FunctorTraits::RunType,
+          typename ArgList = internal::ExtractArgs<RunType>,
+          size_t ArgNum = internal::GetSize<ArgList>,
+          std::enable_if_t<ArgNum == 0>* = nullptr>
+std::vector<ReturnType> CreateVectorParallel(size_t size,
+                                             Generator&& generator) {
+  std::vector<ReturnType> ret(size);
+  OPENMP_PARALLEL_FOR(size_t i = 0; i < size; ++i) { ret[i] = generator(); }
```

Comment:
So `CreateVectorParallel` is only usable when `Generator` is thread-safe right? Maybe this should be added as comment.